### PR TITLE
Pretty formatter options

### PR DESCRIPTION
**Pull request by muhqu**
Adds some new formatting options to the Gherkin Formatter, that are currently pending in a separat PR (https://github.com/cucumber/gherkin/pull/331).
- [x] right-align numeric values in tables (addresses #85, also in PR #105 )
- [x] center-align step keywords
- [x] preserve a single blank line between steps
- [x] preference page for these formatter options

#### How it looks like...
screen shot 2015-02-24 at 09 12 04

cucumber-preferences

#### nice to have future improvements
- allow for project specific preferences via property page

### CODE REVIEW

File: cucumber.eclipse.editor/src/main/java/cucumber/eclipse/editor/editors/GherkinFormatterUtil.java [hunk 1]
```diff
@@ -22,8 +21,14 @@ public static String format(String contents) {
 		// set up
 		StringWriter output = new StringWriter();
 		PrintWriter out = new PrintWriter(output);
-		Formatter formatter = new PrettyFormatter(out, true, false);
+		
+		PrettyFormatter formatter = new PrettyFormatter(out, true, false);
 
+		// TODO: make these configurable preferences!
```
Comment (muhqu) [inline cucumber.eclipse.editor/src/main/java/cucumber/eclipse/editor/editors/GherkinFormatterUtil.java]: :bulb: any help to get these formatter options configurable for the user on a per-project basis would be much appreciated...

File: cucumber.eclipse.editor/src/main/java/cucumber/eclipse/editor/editors/GherkinFormatterUtil.java [hunk 2]
```diff
@@ -22,8 +21,14 @@ public static String format(String contents) {
...
+		formatter.setRightAlignNumericValues(true);
+		formatter.setCenterSteps(true);
+		formatter.setPreserveBlankLineBetweenSteps(true);
+
```
Comment (muhqu) [inline cucumber.eclipse.editor/src/main/java/cucumber/eclipse/editor/editors/GherkinFormatterUtil.java]: :bulb: any help to get these formatter options configurable for the user on a per-project basis would be much appreciated...

**Comment by muhqu** [comment]
@ilanpillemer what do you think? …worth pulling? :shipit:

**Comment by ilanpillemer** [comment]
On Friday, March 13, 2015, Mathias Leppich <notifications@github.com> wrote:

> @ilanpillemer <https://github.com/ilanpillemer> what do you think? …worth
> pulling? [image: :shipit:]
>

In two minds. I don't like separating pretty format logic from gherkin. But
if gherkin is still doing the syntax validation then perhaps it's not so
bad.

**Comment by muhqu** [comment]
> [...] separating pretty format logic from gherkin.

@ilanpillemer I don't know what you mean by that.

If you're referring to the copy of `PrettyFormatter.java` in this PR. It is the same that is waiting to get merged into [gherkin](https://github.com/cucumber/gherkin/). See https://github.com/cucumber/gherkin/pull/331.

**Comment by ilanpillemer** [comment]
On Friday, March 13, 2015, Mathias Leppich <notifications@github.com> wrote:

> [...] separating pretty format logic from gherkin.
>
> @ilanpillemer <https://github.com/ilanpillemer> I don't know what you
> mean by that.
>
> If you're referring to the copy of PrettyFormatter.java in this PR. It is
> the same that is waiting to get merged into gherkin
> <https://github.com/cucumber/gherkin/>. See cucumber/gherkin#331
> <https://github.com/cucumber/gherkin/pull/331>.
>
Ok. I see Aslak says they are all on Gherkin3.

I will merge your changes in over the weekend for this one then seeing this
version of Gherkin is clearly going out anyway.

**Comment by muhqu** [comment]
@ilanpillemer thanks for merge!

**Pull request merged**